packages/shared/src/components/buttons/Button.tsx

```diff
@@ -115,7 +127,14 @@ function ButtonComponent<TagName extends AllowedTags>(
     >
       {icon && getIconWithSize(icon)}
       {children && <span className={spanClassName}>{children}</span>}
-      {rightIcon && getIconWithSize(rightIcon)}
+      {rightIcon && (
+        <ConditionalWrapper
+          condition={rightIconWrapper?.condition}
+          wrapper={rightIconWrapper?.wrapper}
+        >
+          {getIconWithSize(rightIcon)}
+        </ConditionalWrapper>
+      )}
```

Comment:
I still think this is a little bit messy to be honest.
It feels like it abuses the conditional wrapper, what part is this actually needed for?
Maybe we can find a cleaner implementation or revert to rendering it in the component rather.

Reply:
This was some unreverted changes. This was part of using the button before the custom css. Shall revert this.